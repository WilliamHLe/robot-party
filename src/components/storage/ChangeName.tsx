import React, {useState} from "react"
import {setJsonLocal} from "./JsonStorage";
import "./ChangeName.css"

//Komponent som kun vil vises dersom brukeren klikker p책 toggle-knappen "Endre navn"
//Bruker localstorage til 책 lagre et nytt navn fra brukerinput
const ChangeName = () => {

    const [name, setName] = useState("");
    const [saved, setSaved] = useState(false);

    function handleSubmit() {
        //JSON-lagring
        let personalInfo = {
            name: name
        }

        setJsonLocal("personalInfo", personalInfo)
        setSaved(true);

    }


    return (
        <div className={"formWrapper"}>
            <label id="nameInput" htmlFor="name" style={{marginBottom:"10px", marginTop:"5px"}}>Navn:
                <br/>
                <input id="name" type="text" name="Navn" placeholder={name} onChange={e => setName(e.target.value)}/>
            </label>
            <button type="submit" onClick={() => handleSubmit()}>
                Lagre endringer
            </button>
            {saved ? <p>Ditt navn er lagret! Lukk endringsmenyen for 책 f책 ny hilsen.</p> : null}
        </div>
    );
};

export default ChangeName;