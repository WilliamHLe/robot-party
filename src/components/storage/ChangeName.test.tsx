import React from 'react';
import ChangeName from "./ChangeName";
const renderer = require("react-test-renderer");


it('renders correctly', () => {
    const tree = renderer.create(<ChangeName />).toJSON();
    expect(tree).toMatchSnapshot();
});

// test localstorage?