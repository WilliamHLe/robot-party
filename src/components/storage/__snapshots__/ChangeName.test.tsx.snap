// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="formWrapper"
>
  <label
    htmlFor="name"
    id="nameInput"
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "5px",
      }
    }
  >
    Navn:
    <br />
    <input
      id="name"
      name="Navn"
      onChange={[Function]}
      placeholder=""
      type="text"
    />
  </label>
  <button
    onClick={[Function]}
    type="submit"
  >
    Lagre endringer
  </button>
</div>
`;
